src/AddRecordDialog.cpp

```diff
@@ -0,0 +1,247 @@
+#include "AddRecordDialog.h"
+#include "ui_AddRecordDialog.h"
+#include "sqlitetablemodel.h"
+#include "sqlitedb.h"
+#include "Settings.h"
+
+#include <QMessageBox>
+#include <QPushButton>
+#include <QKeyEvent>
+#include <QStyledItemDelegate>
+#include <QWhatsThis>
+#include <QAbstractButton>
+#include <QLineEdit>
+
+// Styled Item Delegate for non-editable columns (all except Value)
+class NoEditDelegate: public QStyledItemDelegate {
+public:
+    NoEditDelegate(QObject* parent=nullptr): QStyledItemDelegate(parent) {}
+    virtual QWidget* createEditor(QWidget *parent, const QStyleOptionViewItem &option, const QModelIndex &index) const {
```

Comment:
I wonder why I don't get these warnings. Is it because of the CMake flags or the GCC version (I'm using 6.3.1)?

> They should probably be silenced 
 
Do you mean through general compiler flags or by other ways that apply only to this particular code?

Reply:
Oh you're probably right. I was using qmake. A quick search didn't yield any results for ```Wall``` or similar in the CMakeLists.txt file. So maybe we don't show as many warning there for some reason.

As for the silencing I meant here in the code, either by removing the name of the parameters (like I do most of the time)
```c++
virtual QWidget* createEditor(QWidget* /*parent*/, const QStyleOptionViewItem& /*option*/, const QModelIndex& /*index*/) const {
```
or by using the ```Q_UNSUED``` macro
```c++
virtual QWidget* createEditor(QWidget *parent, const QStyleOptionViewItem &option, const QModelIndex &index) const {
    Q_UNUSED(parent);
    Q_UNUSED(option);
    Q_UNUSED(index);
```